test/unittests/eof_stack_validation_test.cpp

```diff
@@ -12,145 +12,1214 @@
 using namespace evmone;
 using namespace evmone::test;
 
+namespace
+{
+// code prologue that creates a segment starting with possible stack heights 1 and 3
+const auto varstack = push0() + rjumpi(2, 0) + OP_PUSH0 + OP_PUSH0;
+}  // namespace
+
+TEST_F(eof_validation, unreachable_instructions)
+{
+    add_test_case(
+        eof_bytecode(bytecode{OP_STOP} + OP_STOP), EOFValidationError::unreachable_instructions);
+
+    add_test_case(
+        eof_bytecode(rjump(1) + OP_STOP + OP_STOP), EOFValidationError::unreachable_instructions);
+
+    // STOP reachable only via backwards jump - invalid
+    add_test_case(
+        eof_bytecode(rjump(1) + OP_STOP + rjump(-4)), EOFValidationError::unreachable_instructions);
+}
+
+TEST_F(eof_validation, no_terminating_instruction)
+{
+    add_test_case(eof_bytecode(push0()), EOFValidationError::no_terminating_instruction);
+
+    add_test_case(eof_bytecode(add(1, 2)), EOFValidationError::no_terminating_instruction);
+
+    add_test_case(eof_bytecode(rjumpi(-5, 1)), EOFValidationError::no_terminating_instruction);
+}
+
+TEST_F(eof_validation, non_constant_stack_height)
```

Comment:
Yeah it will break many things, including many relative jumps tests that assume `rjump(offset, 0)` is 5 bytes.

`push0()` was added I think to make possible things like `1023 * push0()`

(this particular test was written even before `push0()`)